src/Action.hs

```diff
@@ -353,18 +353,19 @@ voteIdeaCommentReply ideaId commentId replyId =
 -- FIXME: Compute value in one persistent computation
 markIdeaInJuryPhase :: ActionM m => AUID Idea -> IdeaJuryResultValue -> m ()
 markIdeaInJuryPhase iid rv = do
+    -- FIXME: should this be one transaction?
     idea  <- mquery $ findIdea iid
     topic <- mquery $ ideaTopic idea
-    -- FIXME: should this be one transaction?  or the two above as well?
     equery $ checkInPhase (PhaseJury ==) idea topic
     currentUserAddDb_ (AddIdeaJuryResult iid) rv
     checkCloseJuryPhase topic
 
 checkCloseJuryPhase :: ActionM m => Topic -> m ()
 checkCloseJuryPhase topic = do
+    -- FIXME: should this be one transaction?
```

Comment:
I think so, and the same above. I think an alternative is to check (in the operations above that modify the DB, internally, necessarily within a single transaction with the update) that the current values are as expected, and if not abort with an error like "the ideal is not in the expected phase".